sidecar/src/modelexpress_sidecar/services/downloader.py

```
@@ -0,0 +1,364 @@
+        Download files from S3 using boto3.
+
+        This is a fallback when Model Streamer is not available or for
+        non-SafeTensors files.
+        """
+        import boto3
+        from botocore.config import Config
+
+        scheme, bucket, prefix = self._parse_model_uri(model_path)
+
+        if scheme != "s3":
+            raise ValueError(f"boto3 download only supports S3, got: {scheme}")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)